core/model/upgradeseries.go

```diff
@@ -33,3 +45,16 @@ func ValidateUnitSeriesUpgradeStatus(status UpgradeSeriesStatus) (UpgradeSeriesS
 	}
 	return UpgradeSeriesNotStarted, errors.NotValidf("unit upgrade series status of %q", status)
 }
+
+// CompareUpgradeSeriesStatus compares two upgrade series statuses and returns and integer; if the first
+// argument equals the second then 0 is returned; if the second is greater -1 is
+// returned; 1 is returned otherwise.
+func CompareUpgradeSeriesStatus(status1 UpgradeSeriesStatus, status2 UpgradeSeriesStatus) int {
```

Comment:
Looks like this needs some validation - if one of the status values is not in the map, then the value will be 0. Also we are placing the burden on the caller to interpret what the -1, 0, 1 correspond to.

I was thinking something like:
```go
func (s UpgradeSeriesStatus) ComesBefore(status UpgradeSeriesStatus) (bool, err) {
    i1, ok := UpgradeSeriesStatusOrder[s]
    if !ok {
        return false, errors.Invalidf("status %q", s)
    }

    i2, ok := UpgradeSeriesStatusOrder[status]
    if !ok {
        return false, errors.Invalidf("status %q", status)
    }

    return s < status, nil
}
```
We only really care about not going backwards.

Reply:
I just validated them since a validation function exists already.

Reply:
There is also the general convention is C derived languages of return -1,0,1 from comparison functions.